connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -427,4 +435,62 @@ public void ensureValid(String name, Object value) {
         }
     }
 
+    private static class ResponseHttpHeadersValidator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            String strValue = (String) value;
+            if (strValue.isEmpty()) {
+                return;
+            }
+
+            String[] configs = StringUtil.csvSplit(strValue);
+            Arrays.stream(configs).forEach(config -> validateHttpResponseHeaderConfig(config));
+        }
+
+        private static void validateHttpResponseHeaderConfig(String config) {
```

Comment:
There's no benefit to having the `validateHttpResponseHeaderConfig(...)` method static, so how about making it non-static? And you can use a method reference to make it a tiny bit more readable:
```suggestion
            Arrays.stream(configs).forEach(this::validateHttpResponseHeaderConfig);
        }

        private void validateHttpResponseHeaderConfig(String config) {
```

Or, if you want to make it easier to test, then move to `WorkerConfig` class and make package-level static:
```
            Arrays.stream(configs).forEach(WorkerConfig::validateHttpResponseHeaderConfig);
```
I actually think this is the best way to go, because then you can easily add lots of test methods that thoroughly test each of these methods for both positive and negative cases.